internal/workflow/ProjectAPIKeyOperator.java

```diff
@@ -1,70 +1,76 @@
+        this.projectsApi = new ProjectsApi(client);
     }
 
     public UUID createProjectAPIKey(UUID projectId, String description) throws Exception {
         AuthTokenInput authTokenInput = new AuthTokenInput();
         authTokenInput.setDescription(description);
 
         AuthToken newAuthToken = authApi.createProjectAPIKey(projectId, authTokenInput);
-        UUID newApiKeyId = newAuthToken.getId();
-        return newApiKeyId;
+        return newAuthToken;
     }
 
-    public UUID getOldKeyUUID(UUID projectId, String oldKey) throws Exception {
+    public SimpleEntry<AuthToken, UUID> getProjectAuthTokenByToken(String token) throws Exception {
+        ProjectList projectList = projectsApi.findProjects(null, null, null, null);
+        if (projectList.getProjects().size() != 1) {
+            throw new Exception("Expected only one project to be associated with the token");
+        }
+        UUID projectId = projectList.getProjects().get(0).getId();
+
         AuthTokenList tokenList = authApi.findProjectAPIKeys(projectId, null, null);
-        for (AuthToken token : tokenList.getApiKeys()) {
-            if (token.getToken().equals(oldKey)) {
-                return token.getId();
+        for (AuthToken authToken : tokenList.getApiKeys()) {
+            if (authToken.getToken().equals(token)) {
+                return new SimpleEntry<>(authToken, projectId);
             }
         }
-        throw new Exception("Old API key not found");
+        throw new Exception("Provided token not found");
     }
 
     public void deleteAPIKey(UUID apiKeyId) throws Exception {
         authApi.deleteAPIKey(apiKeyId);
     }
 
-    public static void main(String[] args) {
-        String oldApiKey = "";
-        if (oldApiKey.isEmpty()) {
-            throw new IllegalArgumentException("API Key is not set. Please set the oldApiKey variable.");
-        }
+    public String rotateProjectKey(String authToken) {
+        try {
+            SimpleEntry<AuthToken, UUID> result = getProjectAuthTokenByToken(authToken);
+            AuthToken oldAuthToken = result.getKey();
+            UUID projectId = result.getValue();
 
-        String projectIdString = "";
-        if (projectIdString.isEmpty()) {
-            throw new IllegalArgumentException("Project ID is not set. Please set the projectID variable.");
-        }
+            if (oldAuthToken == null || oldAuthToken.getProject() == null) {
+                throw new Exception("Cannot rotate API key. The current API key is not associated with a project.");
+            }
 
-        ProjectAPIKeyOperator projAPIKeyOperator = new ProjectAPIKeyOperator(oldApiKey);
-        UUID projectId = UUID.fromString(projectIdString);
-        try {
-            projAPIKeyOperator.createProjectAPIKey(projectId, "test-api-key");
+            AuthToken newAuthToken = createProjectAPIKey(projectId, "rotated-api-key");
+            deleteAPIKey(newAuthToken.getId());
```

Comment:
Isn't this deleting the new API key?  It looks like the `deleteAPIKey` function deletes the key whose ID is passed in.  To use the new API key to delete the old one, I think we'd need to reconfigure the API client to use the new API key and then call `deleteAPIKey(oldAuthToken.getId())`.